Lib/test/test_datetime.py

```diff
@@ -20,7 +20,7 @@
 # XXX(gb) First run all the _Pure tests, then all the _Fast tests.  You might
 # not believe this, but in spite of all the sys.modules trickery running a _Pure
 # test last will leave a mix of pure and native datetime stuff lying around.
-test_classes = []
+all_test_classes = []
```

Comment:
> Because the tests were only running for the C implementation, not for the Python
implementation.

This looks like a bug. Please open a new issue for it. This should be fixed in all versions.